src/dscigametrics/metric4.py

```diff
@@ -0,0 +1,20 @@
+def find_best_and_worst_campaigns(df, start_date, end_date, campaign_ids, metric):
+    """
+    Analyzes marketing campaign performance over a specified date range and identifies the best and worst performing campaigns based on a selected metric.
+    This function iterates over a list of campaign IDs, calculates performance metrics for each campaign within the given date range, and then determines the best and worst performing campaigns based on the specified metric.
+
+    Parameters:
```

Comment:
should insert a line like --------- under Parameters. Look at my docstring in compute_metrics.py